### PR TITLE
REF: nanpercentile -> array_algos.quantile

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/array_algos/quantile.py [hunk 1]
```diff
@@ -97,3 +97,92 @@ def quantile_with_mask(
...
+        # Note: we have to do `astype` and not view because in general we
+        #  have float result at this point, not i8
+        return result.astype(values.dtype)
+
+    if not lib.is_scalar(mask) and mask.any():
+        # Caller is responsible for ensuring mask shape match
+        assert mask.shape == values.shape
+        result = [
+            _nanpercentile_1d(val, m, qs, na_value, interpolation=interpolation)
+            for (val, m) in zip(list(values), list(mask))
```
Comment (mroeschke) [inline pandas/core/array_algos/quantile.py:181]: Nit: While refactoring: are the `list` calls necessary here?

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
What's the rationale for this? Currently all nan* functions live in nanops.py, which is a nice pattern?

File: pandas/core/array_algos/quantile.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/array_algos/quantile.py:181]: i wondered the same thing and don't have a good answer.

**Comment by jbrockmendel** [comment]
> What's the rationale for this? Currently all nan* functions live in nanops.py, which is a nice pattern?

these two functions are used exclusively in this file, and don't have any intra-nanops dependencies (e.g. _get_values, _wrap_results).  There's nothing _wrong_ with them being in nanops, but array_algos.quantile is better choice, and remains tightly-scoped.

**Comment by jreback** [comment]
this is fine as it decouples the dependency structure a big. can you rebase

**Comment by jbrockmendel** [comment]
rebased + greenish

**Pull request merged**